modules/directives/src/main/scala/scala/build/preprocessing/directives/ClasspathUtils.scala

```diff
@@ -0,0 +1,7 @@
+package scala.build.preprocessing.directives
+
+object ClasspathUtils {
+  extension (classpathItem: os.Path) {
+    def hasSourceJarSuffix: Boolean = classpathItem.last.endsWith("-sources.jar")
```

Comment:
Is this a new syntax specific to scala-cli, or is it used in another similar tool? If we assume it is added in scala-cli, there should also be a flag to disable it.

Reply:
it's new syntax specific to `scala-cli`, although `*-sources.jar` is a widely used naming scheme for source JARs..
I'd rather not add a flag, as it'd be troublesome to respect it in the `using` directive 😐

Reply:
Indeed, adding a flag may not be a good idea in this case.